### PR TITLE
Terraform 1.0 support

### PR DESCRIPTION
**Pull request by stairsj**
Upgrade the Module to support Terraform 1.x

### CODE REVIEW

**Review by mkhinda29 — APPROVED** [review]
Should we add a .terraform-version file as well?

File: main.tf [hunk 1]
```diff
@@ -14,38 +14,29 @@ resource "aws_cloudfront_origin_access_identity" "default" {
...
-      identifiers = [ aws_cloudfront_origin_access_identity.default.iam_arn ]
+      identifiers = [aws_cloudfront_origin_access_identity.default.iam_arn]
     }
   }
 
   statement {
-    actions   = [ "s3:ListBucket" ]
-    resources = [ "arn:aws:s3:::$${bucket_name}" ]
+    actions   = ["s3:ListBucket"]
+    resources = ["arn:aws:s3:::${local.bucket}${var.origin_path}"]
```
Comment (mkhinda29) [inline main.tf]: Is the extra `${var.origin_path}` added on purpose for least priv?
It's hard to follow without seeing a tf plan, but if bucket_name == local.bucket then it seems the origin path part is new.

**Review by phamtriduy — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (stairsj) [reply main.tf]: good catch, cleaned up.
